moped-editor/src/utils/mapDrawHelpers.js

```diff
@@ -330,255 +116,185 @@ export function useMapDrawTools(
+    setFeatureCollection((prevFeatureCollection) => {
+      const drawnFeaturesWithSourceAndId = makeDrawnFeaturesWithSourceAndId(
+        prevFeatureCollection
+      );
+      /**
+       * Generate a new state including the existing data and any
+       * new features with source and id.
+       */
+      const updatedFeatureCollection = {
+        ...prevFeatureCollection,
+        features: [
+          ...prevFeatureCollection.features,
+          ...drawnFeaturesWithSourceAndId,
+        ],
+      };
+      return updatedFeatureCollection;
+    });
 
     // Dispatch featuresSaved action
     if (saveActionDispatch && runActionDispatch)
       saveActionDispatch({ type: "featuresSaved" });
   };
 
   /**
-   * Takes the click event and sets the draw mode handler and selected mode ID
-   * @param {Object} e - A click event from a draw toolbar button
+   * Callback fired after a single feature is deleted
+   * @param {object} e - Event
    */
-  const switchMode = e => {
-    const switchModeId = e.target.id === modeId ? null : e.target.id;
-    const mode = MODES.find(m => m.id === switchModeId);
-    const currentModeHandler = mode && mode.handler ? new mode.handler() : null;
-
-    // If the button clicked is disableDrawMode, disable drawing...
-    if (mode?.id === "disableDrawMode") setIsDrawing(false);
-    // Else, enable it!
-    else setIsDrawing(true);
-
-    setModeId(switchModeId);
-    setModeHandler(currentModeHandler);
-  };
-
-  /**
-   * Deletes whatever object is selected
-   * https://github.com/uber/nebula.gl/tree/master/modules/react-map-gl-draw#options
-   * onSelect is "a callback when clicking a position when selectable set to true"
-   * @param {object} selected - Holds data about the selected feature
-   */
-  const onSelect = selected => {
-    // Retrieve a list of all features in the map
-    const currentFeatures = mapEditorRef.current.getFeatures();
-    // Remove the feature from the draw UI feature list
-    if (selected.selectedEditHandleIndexes.length) {
-      try {
-        mapEditorRef.current.deleteHandles(
-          selected.selectedFeatureIndex,
-          selected.selectedEditHandleIndexes
-        );
-      } catch (error) {
-        console.log(error.message);
-      }
-      return;
-    }
-
-    if (
-      selected.selectedFeatureIndex === null ||
-      selected.selectedFeatureIndex === undefined
-    ) {
-      return;
-    }
-
-    // Then, remove the feature from the feature collection of the project extent
-    const featureToDelete = currentFeatures[selected.selectedFeatureIndex];
+  const onDelete = (e) => {
+    // Remove the feature from the feature collection of the project extent
+    const featureToDelete = e.features[0];
     const featureIdGetPath = "properties.PROJECT_EXTENT_ID";
     const featureIdToDelete = get(featureToDelete, featureIdGetPath);
 
-    /**
-     * There are duplicates, we need to delete all occurrences of a point.
```

Comment:
This issue with duplicates doesn't seem to be an issue with Mapbox GL Draw. It takes on a lot of the heavy lifting of tracking its own state which is nice for us. 💪